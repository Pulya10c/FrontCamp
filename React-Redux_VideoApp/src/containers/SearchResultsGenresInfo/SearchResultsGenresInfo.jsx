import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import SubSection from "components/SubSection";
import InfoSubTitle from "components/InfoSubTitle";

import { currentVideoGenresSelector } from "selectors/currentVideoSelectors";

const mapStateToProps = state => ({
  genres: currentVideoGenresSelector(state)
});

const SearchResultsGenresInfo = ({ genres }) => {
  return (
    <InfoSubTitle>
      Films by the same {genres.join(", ")}{" "}
      {genres.length === 1 ? "genre" : "genres"}
    </InfoSubTitle>
  );
};

SearchResultsGenresInfo.propTypes = {
  genres: PropTypes.array
};

export default connect(mapStateToProps, null)(SearchResultsGenresInfo);
