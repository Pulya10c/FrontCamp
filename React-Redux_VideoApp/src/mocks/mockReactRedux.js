const createMockMapStateToProps = mapStateToProps => {
  if (!mapStateToProps) {
    return mapStateToProps;
  }

  return (state, ownProps) => {
    return {
      props: mapStateToProps(state, ownProps)
    };
  };
};

const getDispatchFunctionFormShorthandObject = actionsObj => dispatch => {
  return Object.keys(actionsObj).reduce((props, actionKey) => {
    return {
      ...props,
      [actionKey]: function() {
        return dispatch(actionsObj[actionKey].apply(this, arguments));
      }
    };
  }, {});
};

const createMockMapDispatchToProps = mapDispatchToProps => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps;
  }

  const mappingFunction =
    typeof mapDispatchToProps === 'function'
      ? mapDispatchToProps
      : getDispatchFunctionFormShorthandObject(mapDispatchToProps);

  return (dispatch, ownProps) => {
    dispatch = dispatch || jest.fn(action => action);

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    };
  };
};

const createMockMergeProps = mergeProps => {
  if (!mergeProps) {
    return mergeProps;
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: jest.fn(action => action) };

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    };
  };
};

export default {
  connect: (
    mapStateToProps,
    mapDispatchToProps,
    mergeProps
  ) => ConnectedComponent => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    ConnectedComponent
  })
};
