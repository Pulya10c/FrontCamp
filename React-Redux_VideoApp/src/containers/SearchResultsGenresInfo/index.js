import SearchResultsGenresInfo from './SearchResultsGenresInfo';

export default SearchResultsGenresInfo;
