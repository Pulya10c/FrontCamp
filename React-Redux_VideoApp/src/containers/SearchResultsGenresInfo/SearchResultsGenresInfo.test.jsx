import React from "react";
import { shallow } from "enzyme";
import mockReactRedux from "mocks/mockReactRedux";
import SearchResultsGenresInfo from "./";

jest.mock("react-redux", () => mockReactRedux);

const mockGenres = ["mock1", "mock2"];

jest.mock("selectors/currentVideoSelectors", () => ({
  currentVideoGenresSelector: jest.fn(() => mockGenres)
}));

describe("Component SearchResultsGenresInfo", () => {
  const { mapStateToProps, ConnectedComponent } = SearchResultsGenresInfo;

  it("should render correctly", () => {
    const { props } = mapStateToProps();
    expect(shallow(<ConnectedComponent {...props} />)).toMatchSnapshot();
  });

  it("should pass correct props to component from mapStateToProps", () => {
    const { props } = mapStateToProps();
    expect(props).toEqual({
      genres: mockGenres
    });
  });
});
