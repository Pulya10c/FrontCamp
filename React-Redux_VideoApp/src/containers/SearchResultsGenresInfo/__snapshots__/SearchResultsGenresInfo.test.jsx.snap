// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchResultsGenresInfo should render correctly 1`] = `
<styled.h5>
  Films by the same 
  mock1, mock2
   
  genres
</styled.h5>
`;
